src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/AnonymousFactory.php

```diff
@@ -55,7 +55,12 @@ public function getKey()
     public function addConfiguration(NodeDefinition $builder)
     {
         $builder
+            ->beforeNormalization()
+                ->ifTrue(function ($v) { return 'lazy' === $v; })
+                ->then(function ($v) { return ['lazy' => true]; })
+            ->end()
             ->children()
+                ->booleanNode('lazy')->end()
```

Comment:
We'll have to use `empty()` anyway in the extension since the "anonymous" key can be undefined. Which means we can add a default value, but that will change zero bytes in the extension. Not sure it makes sense then.

Reply:
it would still be more explicit for people dumping the config reference.

Reply:
updated